core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,16 +398,58 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            Optional<ConnectorViewDefinition> view = metadata.getView(session, tableName);
+            if (view.isEmpty() && tableHandle.isEmpty()) {
                 throw semanticException(TABLE_NOT_FOUND, showColumns, "Table '%s' does not exist", tableName);
             }
 
+            Relation relation;
+            Expression predicate;
```

Comment:
Could you please extract object?

This will result to less nested code.
Structured code with extracted methods objects is much harder to break and much easier to understand. And much easier to maintain in long term perspective.